ecosystem/indexer/src/default_processor.rs

```diff
@@ -42,23 +43,29 @@ impl Debug for DefaultTransactionProcessor {
 }
 
 fn insert_events(conn: &PgPoolConnection, events: &Vec<EventModel>) {
-    execute_with_better_error(
-        conn,
-        diesel::insert_into(schema::events::table)
-            .values(events)
-            .on_conflict_do_nothing(),
-    )
-    .expect("Error inserting row into database");
+    let chunks = get_chunks(events.len(), EventModel::field_count());
+    for (start_ind, end_ind) in chunks {
+        execute_with_better_error(
+            conn,
+            diesel::insert_into(schema::events::table)
+                .values(&events[start_ind..end_ind])
+                .on_conflict_do_nothing(),
+        )
+        .expect("Error inserting row into database");
+    }
 }
 
 fn insert_write_set_changes(conn: &PgPoolConnection, write_set_changes: &Vec<WriteSetChangeModel>) {
-    execute_with_better_error(
-        conn,
-        diesel::insert_into(schema::write_set_changes::table)
-            .values(write_set_changes)
-            .on_conflict_do_nothing(),
-    )
-    .expect("Error inserting row into database");
+    let chunks = get_chunks(write_set_changes.len(), WriteSetChangeModel::field_count());
```

Comment:
Does field count hit the DB?

Reply:
Nope. It's just a macro that looks at the struct and gets the count of field. No effect on DB.